src/snowflake/cli/api/project/schemas/native_app/application.py

```diff
@@ -14,15 +14,47 @@
 
 from __future__ import annotations
 
-from typing import Optional
+from typing import List, Optional
 
-from pydantic import Field
+from pydantic import Field, model_validator
 from snowflake.cli.api.project.schemas.updatable_model import (
     IdentifierField,
     UpdatableModel,
 )
 
 
+class ApplicationPostDeployHook(UpdatableModel):
+    sql_script: Optional[str] = Field(
+        title="SQL file path relative to the project root", default=None
+    )
+
+    @model_validator(mode="before")
+    @classmethod
+    def ensure_exactly_one_hook_type(cls, hook):
+        supported_keys = cls.model_fields.keys()
+        supported_keys_str = ", ".join(supported_keys)
+        if type(hook) == str:
+            raise ValueError(
+                f"Hooks must be dictionaries with one of the following keys: {supported_keys_str}"
+            )
+        actual_keys = hook.keys()
```

Comment:
can't we add the keys to a set, and ensure that the set size is 1?

Reply:
Hooks can have more than one key. Each hook type has one "primary" key we use to determine its type. For example:
```yml
post_deploy: 
  - sql_script: scripts/post_install.sql
  - grant: execute task
    on: account
```

Reply:
I see. If it can support multiple keys in the future, this will need to be done ad-hoc. Let's discuss offline. I think it might be best to use different classes for different types of hooks.

Reply:
As per our offline suggestion, I agree this is not necessary now. To make this work, we would need something like this:

```
    post_deploy: List[Union[SqlScript, SystemScript]] = Field(
        title="Post application setup ",
        default=None,
        validation_alias="post_deploy",
    )
```
AND
we need to remove ValidationError to SchemaValidationError in UpdatableModel because this is messing with the Union logic.